src/main/java/io/vertx/httpproxy/impl/ReverseProxy.java

```diff
@@ -180,8 +133,25 @@ public ProxyRequest request() {
     public Future<ProxyResponse> sendRequest() {
       if (filters.hasNext()) {
         ProxyInterceptor next = filters.next();
+        if (isWebSocket && !next.allowApplyToWebSocket()) {
```

Comment:
If the next interceptor doesn't support  websockets, shouldn't we only skip it instead of stopping traversal?

Also, what happens if caching is enabled, and the cache interceptor is the first one in line? Can you add a test for this case?

Reply:
Current implementation is "skip it", since the traversal is implemented by recursion.

If caching is enabled, websocket packets will not be cached or operated since CacheFilter will be skipped. I added a test to prove it's functional but I think it's rather hard to test, so not sure if it's good enough.